### PR TITLE
Ecma: increase priority of template_substitution

### PR DESCRIPTION
**Pull request by stsewd**
image

Fixes https://github.com/nvim-treesitter/nvim-treesitter/issues/1688

### CODE REVIEW

File: queries/ecma/highlights.scm [hunk 1]
```diff
@@ -175,9 +175,9 @@
 "{" @punctuation.bracket
 "}" @punctuation.bracket
 
-(template_substitution
-  "${" @punctuation.special
-  "}" @punctuation.special) @none
+((template_substitution ["${" "}"] @punctuation.special) @none
+ ; Substitutios should have more priority over injections.
```
Comment (theHamsta) [inline queries/ecma/highlights.scm:179]: Typo

File: queries/ecma/highlights.scm [hunk 2]
```diff
@@ -175,9 +175,9 @@
...
+ (#set! "priority" 105))
```
Comment (theHamsta) [inline queries/ecma/highlights.scm:180]: But what is then the objective of the injection when it's now shown because it has a lower priority?

Comment (stsewd) [reply queries/ecma/highlights.scm:180]: No sure I understand your question, this only increase the priority of the substitutions (`{foo}`), the injection is still processed (as you can see on the screenshot).

Comment (theHamsta) [reply queries/ecma/highlights.scm:180]: but can you fix the typo?

Comment (stsewd) [reply queries/ecma/highlights.scm:184]: done!

**Review by theHamsta — APPROVED** [review]

**Comment by K1DV5** [comment]
This PR is causing more problems than it solves for me. It works fine when the nesting is only one level deep. But when it's more, like
```javascript
return html`
    <div>
        ${weekDays.map(d => html`<th>${d}</th>`)}
    </div>
`
```
Nothing in the outer `${}` is highlighted. But when I remove the priority part from this commit, Everything except the variables (`weekDays` and `d`) is highlighted properly.

**Comment by theHamsta** [comment]
Priority will be removed in https://github.com/nvim-treesitter/nvim-treesitter/pull/1870 When you remove `(text) @none` from html/highlights.scm` you would see normal highlights, right?

**Comment by K1DV5** [comment]
With `(text) @none` removed from `html_tags/highlights.scm` and `(#set! "priority" 105)` removed from `ecma/highlights.scm`, highlights look normal, except when the `${...}` is in heading tags, in which case it is highlighted like a heading text.

**Comment by K1DV5** [comment]
Also, removing just `@none` instead of `(#set! "priority" 105)` in `ecma/highlights.scm` seems to have the same effect.

**Comment by kyazdani42** [comment]
thing is the html parser gives many errors with js inside braces which produces unexpected results.

**Comment by theHamsta** [comment]
Without HTML injection, and without `priority`
image

With html injection
image

HTML with the now present `(text) @none`
image

**Comment by K1DV5** [comment]
With everything as is
image

With only `(#set! "priority" 105)` removed
image
(Notice that `d` is not highlighted)

With (`(text) @none` or `(#set! "priority" 105)`) and `(#set! "priority" 105)` removed
image

Colorscheme: vscode.nvim

**Comment by oxalica** [comment]
Seems currently we cannot highlight `foo()` when interpolation occurs in some other place.
```js
return html`<div class=${foo()}></div>`
```
`foo` is highlighted with `htmlTSString`, instead of expected `javascriptTSFunction`.

In my mind, interpolation should always be highlighted with the host language. I don't think highlighting would be related to how injected language parse them. It's an issue that html parser **is able to** clear interpolation highlighting.

**Comment by kyazdani42** [comment]
because the html parser does not know that `${...}` is a js interpolation, so it applies things as it should i guess. The thing is, main language should have a higher highlight priority than it's nested parsers.

**Pull request merged**